linter fixes

rfomin opened the pull request.

rfomin commented:
Report with warnings: https://github.com/fabiangreffrath/woof/actions/runs/9313491371/attempts/1#summary-25635970365

fabiangreffrath commented:
> Report with warnings: https://github.com/fabiangreffrath/woof/actions/runs/9313491371/attempts/1#summary-25635970365

That's cool!

rfomin commented on .github/workflows/main.yml:
```diff
@@ -178,7 +178,24 @@ jobs:
+            ,clang-analyzer-*
+            ,-clang-analyzer-cplusplus*
+            ,-clang-analyzer-security*
+            ,-clang-analyzer-valist*
```
Iy seems `valist` checks are buggy: https://github.com/llvm/llvm-project/issues/40656

fabiangreffrath replied:
So, are you going disable them?

rfomin replied:
I already disabled them. `-clang-analyzer-*` means we enable all `analyzer` checks, `-clang-analizer-valist*` means we disable all `valist` checks.

fabiangreffrath replied:
Ah, I overlooked the `-` prefix to the command.

fabiangreffrath commented on src/mn_menu.c:
```diff
@@ -1522,8 +1522,6 @@ static void M_EndGame(int choice)
 static void M_ChangeMessages(int choice)
 {
-    // warning: unused parameter `int choice'
-    choice = 0;
```
This was once added to fix another warning. I think we could just invalidate the variable by calling `(void)choice;`.

rfomin replied:
Is this a "pedantic" warning? I think both warnings are unnecessary, we can disable this clang-tidy check (although it did find some dead code).

fabiangreffrath commented on .github/workflows/main.yml:
```diff
@@ -178,7 +178,24 @@ jobs:
+            ,-clang-analyzer-security*
+            ,-clang-analyzer-valist*
+          database: 'build'
+          ignore: 'miniz|spng|win32|toolsrc|src/i_pcsound.c'
```
Why do we exclude `src/i_pcsound.c`?

rfomin replied:
i_pcsound.c only works if `HAVE_AL_BUFFER_CALLBACK` is defined (ubuntu-24.04 doesn't have it). Although clang-tidy supports a compiler database, it lacks any file filtering, so we have to filter it manually.

rfomin commented:
I've fixed most of the warnings, the rest are not so trivial or related to padding (I don't think it's worth rearranging those structs). So I removed the workflow failing, maybe someone will fix the warnings one day.

See the full report in the workflow summary: https://github.com/fabiangreffrath/woof/actions/runs/9606782914/attempts/1#summary-26496852091

fabiangreffrath reviewed (approved):
Fine with me, thanks!
